sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -110,6 +111,11 @@ private[sql] class SharedState(
    */
   val cacheManager: CacheManager = new CacheManager
 
+  /**
+   * Class for caching streaming query instance.
+   */
+  val streamingQueryCache = new mutable.HashSet[(StreamingQuery, Long)]
```

Comment:
In other words, it might be ideal to create a new class to serve this. We can add some of methods to abstract the functionality and caller wouldn't need to know about the details of cache (the underlying data structure). In addition, we can add tests for the new class.

Reply:
make sense